src/App.tsx

```diff
@@ -1,223 +1,200 @@
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import classNames from 'classnames';
 import { AuthContext } from './components/Auth/AuthContext';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { client } from './utils/fetchClient';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+import { TodosFilter, TodoError } from './types/TodoFilter';
+import { ErrorInfo } from './components/ErrorInfo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [formInput, setFormInput] = useState('');
+  const [filter, setFilter] = useState<TodosFilter>(TodosFilter.all);
+  const [error, setError] = useState<TodoError>(TodoError.noerror);
+  const [isAdding, setIsAdding] = useState(false);
+
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const getTodosFromServer = () => {
+    getTodos(user?.id)
+      .then(setTodos);
+  };
+
+  const todosFiltration = (filterType: TodosFilter) => {
+    switch (filterType) {
+      case TodosFilter.active:
+        setVisibleTodos(todos.filter(todo => todo.completed === false));
+        setFilter(TodosFilter.active);
+        break;
+      case TodosFilter.completed:
+        setVisibleTodos(todos.filter(todo => todo.completed === true));
+        setFilter(TodosFilter.completed);
+        break;
+      default:
+        setVisibleTodos(todos);
+        setFilter(TodosFilter.all);
+    }
+  };
+
+  const errorTimeout = () => {
+    setTimeout(() => {
+      setError(TodoError.noerror);
+    }, 3000);
+  };
+
+  const errorHandler = (errorType: TodoError) => {
+    switch (errorType) {
+      case TodoError.delete:
+        setError(TodoError.delete);
+        errorTimeout();
+        break;
+      case TodoError.update:
+        setError(TodoError.update);
+        errorTimeout();
+        break;
+      case TodoError.add:
+        setError(TodoError.add);
+        errorTimeout();
+        break;
+      case TodoError.empty:
+        setError(TodoError.empty);
+        errorTimeout();
+        break;
+      default:
+        setError(TodoError.noerror);
+    }
+  };
+
+  const errorButtonHandler = () => {
+    setError(TodoError.noerror);
+  };
+
+  const submitHandler = async (event: React.FormEvent) => {
+    event.preventDefault();
+    setIsAdding(true);
+    errorHandler(TodoError.noerror);
+    const date = {
+      userId: user?.id,
+      title: formInput,
+      completed: false,
+    };
+
+    if (formInput !== '') {
+      try {
+        await client.post('/todos', date);
+      } catch {
+        errorHandler(TodoError.add);
+      }
+    } else {
+      setError(TodoError.empty);
+      errorHandler(TodoError.empty);
+    }
+
+    setTimeout(() => {
```

Comment:
why do that? If a person has a slow Internet, then the data will not have time to update, as you are already asking for them again. Also, it is not efficient to do it every time when submitting, even if the input field is empty. 

1. Check your post method. if it returns the posted value, then just add this value to the todos array, if not, then only after updating the data make a request (line 93)
2.  `if (!formInput.trim())` - so you can avoid the phenomenon where I can add empty spaces and save todo